mobile/native/src/ln_dlc/lightning_subscriber.rs

```diff
@@ -0,0 +1,35 @@
+use crate::event;
+use crate::event::EventInternal;
+use crate::ln_dlc::node::Node;
+use lightning::util::events::Event;
+use tokio::sync::watch::Receiver;
+
+impl Node {
+    pub async fn listen_for_lightning_events(&self, mut event_receiver: Receiver<Option<Event>>) {
+        loop {
+            let event = match event_receiver.changed().await {
+                Ok(()) => {
+                    if let Some(event) = event_receiver.borrow().clone() {
+                        event
+                    } else {
+                        continue;
+                    }
+                }
+                Err(e) => {
+                    tracing::error!("Failed to receive event: {e:#}");
+                    continue;
+                }
+            };
+
+            match event {
```

Comment:
🙃 If you move this `match` inside of the `if let` above then you don't need to use `continue`s.